src/arch/riscv/pmp/pmp.c

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file
+ *
+ * @brief RISC-V Physical Memory Protection (PMP) Driver
+ *
+ * @date 02.08.2024
+ * @author Suraj Ravindra Sonawane
+ */
+
+#include "pmp.h"
+#include <math.h>
+#include <hal/reg.h>
+
+/**
+ * Writes to a PMP configuration register.
+ *
+ * @param reg The index of the PMP configuration register to write to.
+ * @param value The value to write to the register.
+ */
+static inline void write_pmpcfg(uint32_t reg, uint32_t value) {
+    REG32_STORE(PMP_CFG_BASE + reg * 4, value);
+}
+
+/**
+ * Writes to a PMP address register.
+ *
+ * @param reg The index of the PMP address register to write to.
+ * @param value The address value to write to the register.
+ */
+static inline void write_pmpaddr(uint32_t reg, uint32_t value) {
+    REG32_STORE(PMP_ADDR_BASE + reg * 4, value);
+}
+
+/**
+ * Initializes the PMP subsystem by clearing all PMP entries and setting up 
+ * predefined PMP regions based on configuration options.
+ *
+ */
+int pmp_init(void) {
```

Comment:
Does it call with UNIT_INIT or directly from code? If it call from code where it will be?

Reply:
Sorry, I forgot to call it, but I will do it now using UNIT_INIT.